plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/transport/BatchTask.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.arrow.flight.transport;
+
+import org.opensearch.Version;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.transport.TransportResponse;
+import org.opensearch.transport.TcpChannel;
+
+import java.util.Set;
+
+/**
+ * Represents a batch processing task for Flight streaming responses.
+ * @opensearch.internal
+ */
+class BatchTask implements AutoCloseable {
```

Comment:
Can you use a record class here to reduce a bunch of this boilerplate?

Reply:
done